openmc/material.py

```diff
@@ -523,6 +523,95 @@ def remove_element(self, element):
             if element == elm:
                 self._elements.remove(elm)
 
+    def add_material(self, material, percent, percent_type='ao'):
+        """Add a material to the material
+
+        Parameters
+        ----------
+        material : openmc.Material
+            Material to add
+        percent : float
+            Atom or weight percent
+        percent_type : {'ao', 'wo'}
+            'ao' for atom percent and 'wo' for weight percent
+
+        """
+
+        if self._macroscopic is not None:
+            msg = 'Unable to add a Material to Material ID="{0}" as a ' \
+                  'macroscopic data-set has already been added'.format(self._id)
```

Comment:
Consider using `self.id` instead of `self._id` here and elsewhere. I don't think we are perfectly consistent, but it is a goal to use the properties rather than reference attributes directly.